api/uniter/state_test.go

```diff
@@ -74,3 +70,15 @@ func (s *stateSuite) TestAllMachinePorts(c *gc.C) {
 		{1, 8, "udp"}:     {Unit: wordpressUnit1.Tag().String()},
 	})
 }
+
+func (s *stateSuite) assertOpenPort(c *gc.C, u *state.Unit, subnet, protocol string, port int) {
+	s.assertOpenPorts(c, u, subnet, protocol, port, port)
+}
+
+func (s *stateSuite) assertOpenPorts(c *gc.C, u *state.Unit, subnet, protocol string, from, to int) {
```

Comment:
That's actually a nice idea. I will push a commit to extract these helpers to a single place.

Reply:
Addressed in https://github.com/juju/juju/pull/11204/commits/db477d5bc16451135f8e62981e3390e654293cc3. I changed the unit.OpenClosePortsInSubnet signature to work with `network.PortRange` slices to avoid import cycles while importing the mixing in internal state tests.